### PR TITLE
echidna: 2.2.2 -> 2.2.3

### PR DESCRIPTION
**Pull request by hellwolf**
## Description of changes

Upgrade to 2.2.3 with the latest haskellPackages.

The used patch can be found at: https://github.com/crytic/echidna/pull/1290

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by hellwolf** [comment]
@ofborg build echidna

**Comment by hellwolf** [comment]
deferring 2.2.4 update, due to nixpkgs's hevem is incompatible.

File: pkgs/tools/security/echidna/default.nix [hunk 1]
```diff
@@ -1,53 +1,107 @@
...
     repo = "echidna";
     rev = "v${version}";
-    sha256 = "sha256-l1ILdO+xb0zx/TFM6Am9j5hq1RnIMNf2HU6YvslAj0w=";
+    sha256 = "sha256-NJ2G6EkexYE4P3GD7PZ+lLEs1dqnoqIB2zfAOD5SQ8M=";
   };
 
-  isLibrary = true;
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/crytic/echidna/compare/v2.2.3...hellwolf:echidna:2913b02.patch";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/echidna/default.nix:23]:
```suggestion
      url = "https://github.com/crytic/echidna/commit/2913b027d7e793390ed489ef6a47d23ec9b3c800.patch";
```

File: pkgs/tools/security/echidna/default.nix [hunk 2]
```diff
@@ -1,53 +1,107 @@
...
     owner = "crytic";
     repo = "echidna";
     rev = "v${version}";
-    sha256 = "sha256-l1ILdO+xb0zx/TFM6Am9j5hq1RnIMNf2HU6YvslAj0w=";
+    sha256 = "sha256-NJ2G6EkexYE4P3GD7PZ+lLEs1dqnoqIB2zfAOD5SQ8M=";
   };
 
-  isLibrary = true;
+  patches = [
+    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/tools/security/echidna/default.nix:22]:
```suggestion
    # Support cross platform vty 6.x with vty-crossplatform
    # https://github.com/crytic/echidna/pull/1290
    (fetchpatch {
```

File: pkgs/tools/security/echidna/default.nix (hunk 1 above)
Comment (hellwolf) [reply pkgs/tools/security/echidna/default.nix]: Okay, sure. I thought the tag was "immutable" enough.

**Comment by SuperSandro2000** [comment]
@øfborg eval

**Comment by hellwolf** [comment]
huh, how the eval failed?

**Comment by hellwolf** [comment]
https://github.com/ofborg eval

**Comment by hellwolf** [comment]
@ofborg build echidna

**Pull request merged**